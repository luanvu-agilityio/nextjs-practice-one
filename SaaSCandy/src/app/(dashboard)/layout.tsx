import { ReactNode } from 'react';

interface DashboardLayoutProps {
  children: ReactNode;
}

const DashboardLayout = ({ children }: Readonly<DashboardLayoutProps>) => {
  return <div>{children}</div>;
};

export default DashboardLayout;
