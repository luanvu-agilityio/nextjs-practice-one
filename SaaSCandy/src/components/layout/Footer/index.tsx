'use client';

import { Button, Heading, IconButton, Typography } from '@/components/common';
import LogoIcon from '@/components/icons/Logo';
import { NAV_LINKS, SOCIAL_LINKS } from '@/constants';

import Link from 'next/link';
import { JSX } from 'react';

export default function Footer(): JSX.Element {
  return (
    <footer className='bg-dark-blue w-full '>
      <div className='max-w-[1296px] mx-auto px-0 py-17.5 flex justify-between gap-33'>
        <div className='w-76.5 flex flex-col gap-13.5'>
          <Link href='/' className='flex items-center gap-2 no-underline'>
            <LogoIcon className='w-8 h-8' />

            <div className='hidden sm:block'>
              <Heading
                as='h1'
                size='md'
                content={
                  <span className='font-secondary text-white text-3xl font-light'>
                    SaaS<span className='font-medium'>Candy</span>
                  </span>
                }
              />
            </div>
          </Link>
          <div className='flex flex-col gap-6.5'>
            <Typography className='text-lg text-white opacity-50 font-medium'>
              Rakon is a simple, elegant, and secure way to build your bitcoin
              and crypto portfolio.
            </Typography>
            <Typography className='text-white text-lg font-medium'>
              1989 Don Jackson Lane
              <br />
              <span className='text-lg text-blue-foreground font-medium'>
                Call us:{' '}
                <span className='text-orange-background'>808-956-9599</span>
              </span>
            </Typography>
          </div>
        </div>
        <div className='flex justify-between  gap-6 text-lg '>
          <div className='w-48'>
            <Heading
              as='h4'
              content='Services'
              className='text-white mb-10 font-medium text-lg'
            />
            <ul className='space-y-6 text-lg font-medium text-white hover:text-white opacity-50'>
              <li>
                <Link href='/services/edtech' className='hover:text-white'>
                  EdTech Apps
                </Link>
              </li>
              <li>
                <Link href='/services/ecommerce' className='hover:text-white'>
                  eCommerce Apps
                </Link>
              </li>
              <li>
                <Link href='/services/crm' className='hover:text-white'>
                  CRM Apps
                </Link>
              </li>
              <li>
                <Link href='/services/health' className='hover:text-white'>
                  Health Apps
                </Link>
              </li>
              <li>
                <Link href='/services/analytics' className='hover:text-white'>
                  Web Analytics Apps
                </Link>
              </li>
              <li>
                <Link href='/services/banking' className='hover:text-white'>
                  Banking Apps
                </Link>
              </li>
            </ul>
          </div>

          <div className='w-48'>
            <Heading
              as='h4'
              content='Company'
              className='text-white text-lg mb-10 font-medium '
            />
            <ul className='space-y-6 text-lg font-medium text-white opacity-50 hover:text-white'>
              {NAV_LINKS.map(l => (
                <li key={l.href}>
                  <Link href={l.href} className='hover:text-white'>
                    {l.label}
                  </Link>
                </li>
              ))}
            </ul>
          </div>

          <div className='w-104'>
            <Heading
              as='h4'
              content='Subscribe'
              className='text-white mb-10 font-medium text-lg'
            />
            <Typography className='text-lg font-medium text-white opacity-50 mb-4'>
              Subscribe to get the latest news from us
            </Typography>
            <form className='flex gap-2.5'>
              <input
                aria-label='Email'
                type='email'
                placeholder='Enter email address'
                className='flex-1 rounded-lg bg-transparent border border-gray-700 px-3 py-2 text-sm placeholder-gray-400'
              />
              <Button variant='primary' size='small'>
                Register
              </Button>
            </form>
          </div>
        </div>
      </div>

      <div className='relative max-w-[1296px] mx-auto pb-17.5'>
        <div className='text-lg font-medium text-white flex items-center justify-between'>
          <span className='opacity-50'>
            Â©{new Date().getFullYear()} - All Rights Reserved by GetNextJs
          </span>

          <div className='flex gap-6' aria-label='Social media links'>
            {SOCIAL_LINKS.map(s => {
              const IconComponent = s.icon;
              return (
                <IconButton key={s.href} variant='ghost'>
                  <Link
                    href={s.href}
                    target='_blank'
                    rel='noreferrer'
                    className='inline-block'
                  >
                    <IconComponent />
                  </Link>
                </IconButton>
              );
            })}
          </div>
        </div>
      </div>

      <Link
        href='#'
        className='fixed right-10 bottom-10 bg-orange-background text-white text-md font-semibold px-4 py-2.5 rounded-xl'
      >
        Get This Template
      </Link>
    </footer>
  );
}
