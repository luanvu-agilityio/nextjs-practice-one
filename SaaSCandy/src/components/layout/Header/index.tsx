'use client';

import { JSX, useState } from 'react';
import Link from 'next/link';
import { useSession } from '@/lib/auth-client';
import { User } from 'lucide-react';

// Icons
import { LogoIcon, ThemeSwitcherIcon } from '@/icons';

// Components
import Navbar from '@/components/NavBar';
import { Button, Heading } from '@/components/common';
import { IconButton } from '@/components/common/IconButton';
import { UserMenu } from '@/components/UserMenu';
import AuthSectionSkeleton from './AuthSectionSkeleton';

// Constants
import { ROUTES } from '@/constants';

const Header = (): JSX.Element => {
  const { data: session, isPending } = useSession();
  const [showMobileAuth, setShowMobileAuth] = useState(false);

  const renderAuthSection = () => {
    if (isPending) {
      return <AuthSectionSkeleton />;
    }

    if (session?.user) {
      return <UserMenu />;
    }

    return (
      <>
        {/* Desktop Auth Buttons */}
        <div className='hidden sm:flex items-center gap-2 sm:gap-4'>
          <Link href={ROUTES.SIGN_IN} className='no-underline'>
            <Button variant='secondary' size='small'>
              Sign In
            </Button>
          </Link>
          <Link href={ROUTES.SIGN_UP} className='no-underline'>
            <Button variant='primary' size='small'>
              Sign Up
            </Button>
          </Link>
        </div>

        {/* Mobile User Icon Toggle */}
        <div className='lg:hidden relative'>
          <IconButton
            aria-label='Toggle auth menu'
            variant='ghost'
            size='md'
            onClick={() => setShowMobileAuth(!showMobileAuth)}
            className={`text-primary ${showMobileAuth ? 'bg-gray-100' : ''}`}
          >
            <User className='w-6 h-6' />
          </IconButton>

          {/* Mobile Auth Dropdown */}
          {showMobileAuth && (
            <>
              {/* Backdrop to close on click outside */}
              <div
                className='fixed inset-0 z-30'
                onClick={() => setShowMobileAuth(false)}
              />

              {/* Dropdown Menu */}
              <div className='absolute right-0 top-full mt-2 w-48 bg-white rounded-lg shadow-lg border border-gray-200 py-2 z-40'>
                <Link
                  href={ROUTES.SIGN_IN}
                  className='no-underline block'
                  onClick={() => setShowMobileAuth(false)}
                >
                  <Button
                    variant='secondary'
                    size='small'
                    className='w-full mx-2 mb-2'
                    style={{ width: 'calc(100% - 1rem)' }}
                  >
                    Sign In
                  </Button>
                </Link>
                <Link
                  href={ROUTES.SIGN_UP}
                  className='no-underline block'
                  onClick={() => setShowMobileAuth(false)}
                >
                  <Button
                    variant='primary'
                    size='small'
                    className='w-full mx-2'
                    style={{ width: 'calc(100% - 1rem)' }}
                  >
                    Sign Up
                  </Button>
                </Link>
              </div>
            </>
          )}
        </div>
      </>
    );
  };

  return (
    <header className='bg-white w-full'>
      <div className='max-w-[1296px] mx-auto py-4 px-4 sm:px-6 lg:px-0 flex items-center gap-2 sm:gap-4 lg:gap-14'>
        <Link
          href='/'
          className='flex items-center gap-2 no-underline flex-shrink-0'
          aria-label='Homepage'
        >
          <LogoIcon className='w-6 h-6 sm:w-8 sm:h-8' />
          <span className='sr-only'>Homepage</span>
          <div className='hidden sm:block'>
            <Heading
              as='h1'
              size='md'
              content={
                <span className='font-secondary text-primary text-xl sm:text-2xl lg:text-3xl font-light'>
                  SaaS<span className='font-medium'>Candy</span>
                </span>
              }
            />
          </div>
        </Link>

        <Navbar />

        <div className='ml-auto flex items-center gap-2 sm:gap-4'>
          <ThemeSwitcherIcon className='w-8 h-8 sm:w-10 sm:h-10' />

          {/* Auth section with toggle on mobile */}
          {renderAuthSection()}
        </div>
      </div>
    </header>
  );
};

Header.displayName = 'Header';
export { Header };
