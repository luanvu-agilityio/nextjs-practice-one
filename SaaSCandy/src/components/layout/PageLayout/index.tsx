'use client';

import { Fragment, useState } from 'react';
import { usePathname } from 'next/navigation';
import Link from 'next/link';
import { useSession } from '@/lib/auth-client';
import { User } from 'lucide-react';

// Icons
import { LogoIcon, ThemeSwitcherIcon } from '@/icons';

// Components
import { Heading, Typography, Button } from '@/components/common';
import { IconButton } from '@/components/common/IconButton';
import Navbar from '@/components/NavBar';
import { UserMenu } from '@/components/UserMenu';
import AuthSectionSkeleton from '../Header/AuthSectionSkeleton';
import {
  Breadcrumb,
  BreadcrumbList,
  BreadcrumbItem,
  BreadcrumbLink,
  BreadcrumbSeparator,
  BreadcrumbPage,
} from '@/components/common/Breadcrumb';

// Utils
import { extractBreadcrumbs } from '@/utils';
import { ROUTES } from '@/constants';

interface PageLayoutProps {
  title: string;
  subtitle?: string;
  children: React.ReactNode;
  breadcrumbOverride?: Array<{
    label: string;
    href: string;
    isActive: boolean;
  }>;
  className?: string;
  headerClassName?: string;
  contentClassName?: string;
}

const PageLayout = ({
  title,
  subtitle,
  children,
  breadcrumbOverride,
  className = '',
  headerClassName = '',
  contentClassName = '',
}: PageLayoutProps) => {
  const pathname = usePathname();
  const breadcrumbs = breadcrumbOverride || extractBreadcrumbs(pathname);
  const { data: session, isPending } = useSession();
  const [showMobileAuth, setShowMobileAuth] = useState(false);

  const renderAuthSection = () => {
    if (isPending) {
      return <AuthSectionSkeleton />;
    }

    if (session?.user) {
      return <UserMenu />;
    }

    return (
      <>
        {/* Desktop Auth Buttons */}
        <div className='hidden sm:flex items-center gap-2 sm:gap-4'>
          <Link href={ROUTES.SIGN_IN} className='no-underline'>
            <Button variant='secondary' size='small'>
              Sign In
            </Button>
          </Link>
          <Link href={ROUTES.SIGN_UP} className='no-underline'>
            <Button variant='primary' size='small'>
              Sign Up
            </Button>
          </Link>
        </div>

        {/* Mobile User Icon Toggle */}
        <div className='sm:hidden relative'>
          <IconButton
            aria-label='Toggle auth menu'
            variant='ghost'
            size='md'
            onClick={() => setShowMobileAuth(!showMobileAuth)}
            className={`text-primary p-0 ${showMobileAuth ? 'bg-gray-100' : ''}`}
          >
            <User className='w-6 h-6' />
          </IconButton>

          {/* Mobile Auth Dropdown */}
          {showMobileAuth && (
            <>
              {/* Backdrop to close on click outside */}
              <div
                className='fixed inset-0 z-30'
                onClick={() => setShowMobileAuth(false)}
              />

              {/* Dropdown Menu */}
              <div className='absolute right-0 top-full mt-2 w-48 bg-white rounded-lg shadow-lg border border-gray-200 py-2 z-40'>
                <Link
                  href={ROUTES.SIGN_IN}
                  className='no-underline block'
                  onClick={() => setShowMobileAuth(false)}
                >
                  <Button
                    variant='secondary'
                    size='small'
                    className='w-full mx-2 mb-2'
                    style={{ width: 'calc(100% - 1rem)' }}
                  >
                    Sign In
                  </Button>
                </Link>
                <Link
                  href={ROUTES.SIGN_UP}
                  className='no-underline block'
                  onClick={() => setShowMobileAuth(false)}
                >
                  <Button
                    variant='primary'
                    size='small'
                    className='w-full mx-2'
                    style={{ width: 'calc(100% - 1rem)' }}
                  >
                    Sign Up
                  </Button>
                </Link>
              </div>
            </>
          )}
        </div>
      </>
    );
  };

  return (
    <div className={`${className}`}>
      {/* Hero Header Section with integrated Header */}
      <div
        className={`relative text-white ${headerClassName}`}
        style={{
          backgroundImage: "url('/images/background/heading-background.png')",
          backgroundSize: 'cover',
        }}
      >
        {/* Header Navigation */}
        <header className='w-full relative z-10'>
          <div className='max-w-[1296px] mx-auto py-4 px-4 sm:px-6 lg:px-0 flex items-center gap-4 sm:gap-8 lg:gap-14'>
            <Link
              href='/'
              className='flex items-center gap-2 no-underline flex-shrink-0'
            >
              <LogoIcon className='w-6 h-6 sm:w-8 sm:h-8' />
              <div className='hidden sm:block'>
                <Heading
                  as='h1'
                  size='md'
                  content={
                    <span className='font-secondary text-primary text-xl sm:text-2xl lg:text-3xl font-light'>
                      SaaS<span className='font-medium'>Candy</span>
                    </span>
                  }
                />
              </div>
            </Link>

            <Navbar />

            <div className='ml-auto flex items-center gap-2 sm:gap-4'>
              <ThemeSwitcherIcon className='w-8 h-8 sm:w-10 sm:h-10' />
              {renderAuthSection()}
            </div>
          </div>
        </header>

        {/* Page Content */}
        <div className='relative flex flex-col lg:flex-row items-start lg:items-center justify-between max-w-[1296px] mx-auto px-4 sm:px-6 lg:px-0 pb-12 sm:pb-16 pt-12 sm:pt-20 lg:pt-29.25 gap-8'>
          {/* Left Content */}
          <div className='space-y-4 sm:space-y-6 lg:space-y-8 w-full lg:w-auto'>
            <Heading
              as='h2'
              size='xl'
              content={title}
              className='text-primary font-bold text-3xl sm:text-4xl lg:text-5xl xl:text-6xl'
            />
            {subtitle && (
              <Typography
                size='lg'
                content={subtitle}
                className='text-gray-background leading-relaxed max-w-full lg:max-w-132 sm:text-lg'
              />
            )}
          </div>

          {/* Right Breadcrumb */}
          <Breadcrumb className='w-full lg:w-auto lg:flex-shrink-0'>
            <BreadcrumbList className='flex-wrap'>
              {breadcrumbs.map((item, index) => (
                <Fragment key={item.label}>
                  <BreadcrumbItem>
                    {item.isActive ? (
                      <BreadcrumbPage className='text-white font-medium text-sm sm:text-xs'>
                        {item.label}
                      </BreadcrumbPage>
                    ) : (
                      <BreadcrumbLink
                        href={item.href}
                        className='text-blue-foreground hover:text-white transition-colors text-sm sm:text-xs'
                      >
                        {item.label}
                      </BreadcrumbLink>
                    )}
                  </BreadcrumbItem>
                  {index < breadcrumbs.length - 1 && (
                    <BreadcrumbSeparator className='text-white' />
                  )}
                </Fragment>
              ))}
            </BreadcrumbList>
          </Breadcrumb>
        </div>
      </div>

      {/* Main Content */}
      <div className={`bg-inactive-background ${contentClassName}`}>
        {children}
      </div>
    </div>
  );
};

PageLayout.displaName = 'PageLayout';
export { PageLayout };
