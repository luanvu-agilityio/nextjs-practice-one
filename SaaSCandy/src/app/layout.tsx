'use client';

// import type { Metadata } from 'next';
import { usePathname } from 'next/navigation';

// Google Fonts
import { Manrope, Ubuntu } from 'next/font/google';

// CSS variables
import './globals.css';

// Providers
import {
  ReactQueryProvider,
  SessionProvider,
  ToastProvider,
} from '@/providers';

// Components
import Header from '@/components/layout/Header';
import Footer from '@/components/layout/Footer';

const manrope = Manrope({
  variable: '--font-primary',
  subsets: ['latin'],
  display: 'swap',
});

const ubuntu = Ubuntu({
  variable: '--font-secondary',
  subsets: ['latin'],
  weight: ['300', '500'],
  display: 'swap',
});

// Pages that use PageLayout (don't need separate Header)
const PAGE_LAYOUT_ROUTES = [
  '/portfolio',
  '/pricing',
  '/services',
  '/docs',
  '/blog',
  '/contact',
];

// export const metadata: Metadata = {
//   generator: 'Next.js',
//   title: 'SaaSCandy',
//   description: 'Build Innovative Apps For Your Business',
//   keywords: ['SaaS', 'Candy', 'Business', 'Apps', 'Innovation'],
//   authors: [{ name: 'Luan Vu' }],
//   viewport: 'width=device-width, initial-scale=1',
//   robots: 'index, follow',
//   openGraph: {
//     title: 'SaaSCandy',
//     description: 'Build Innovative Apps For Your Business',
//     url: 'https://saascandy.com',
//     siteName: 'SaaSCandy',
//     images: [
//       {
//         url: 'https://saascandy.com/og-image.png',
//         width: 1200,
//         height: 630,
//         alt: 'SaaSCandy Logo',
//       },
//     ],
//     locale: 'en_US',
//     type: 'website',
//   },
// };

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  const pathname = usePathname();
  const shouldShowHeader =
    pathname !== '/' &&
    (!PAGE_LAYOUT_ROUTES.some(
      route =>
        pathname === route || (route !== '/blog' && pathname?.startsWith(route))
    ) ||
      (pathname?.startsWith('/blog/') && pathname !== '/blog'));

  return (
    <html lang='en' suppressHydrationWarning className='overflow-x-hidden'>
      <body className={`${ubuntu.variable} ${manrope.variable} antialiased `}>
        <ReactQueryProvider>
          <SessionProvider>
            <ToastProvider>
              {shouldShowHeader && <Header />}
              {children}
              <Footer />
            </ToastProvider>
          </SessionProvider>
        </ReactQueryProvider>
      </body>
    </html>
  );
}
