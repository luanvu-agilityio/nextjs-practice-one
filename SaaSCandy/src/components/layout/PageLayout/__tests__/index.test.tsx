import { render, screen, fireEvent } from '@testing-library/react';
import { PageLayout } from '@/components/layout/PageLayout';
import { useSession } from '@/lib/auth-client';
import { usePathname } from 'next/navigation';

jest.mock('@/lib/auth-client');
jest.mock('next/navigation', () => ({
  usePathname: jest.fn(),
  useRouter: jest.fn(),
}));
jest.mock('@/utils', () => ({
  extractBreadcrumbs: jest.fn(() => [
    { label: 'Home', href: '/', isActive: false },
    { label: 'Services', href: '/services', isActive: true },
  ]),
}));
jest.mock('@/constants', () => ({
  ROUTES: {
    SIGN_IN: '/signin',
    SIGN_UP: '/signup',
  },
}));

jest.mock('@/constants', () => ({
  ROUTES: {
    SIGN_IN: '/signin',
    SIGN_UP: '/signup',
  },
  NAV_LINKS: [
    { label: 'Home', href: '/' },
    { label: 'Services', href: '/services' },
    { label: 'About', href: '/about' },
    { label: 'Contact', href: '/contact' },
    { label: 'Blog', href: '/blog' },
  ],
}));

const mockUseSession = useSession as jest.MockedFunction<typeof useSession>;
const mockUsePathname = usePathname as jest.MockedFunction<typeof usePathname>;

describe('PageLayout - Snapshot Tests', () => {
  beforeEach(() => {
    mockUseSession.mockReturnValue({
      data: null,
      isPending: false,
      error: null,
      refetch: function (): void {
        throw new Error('Function not implemented.');
      },
      isRefetching: false,
    });
    mockUsePathname.mockReturnValue('/services');
  });

  it('should match snapshot', () => {
    const { container } = render(
      <PageLayout title='Test Page' subtitle='Test subtitle'>
        <div>Test Content</div>
      </PageLayout>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render title and subtitle', () => {
    render(
      <PageLayout title='Test Page' subtitle='Test subtitle'>
        <div>Test Content</div>
      </PageLayout>
    );

    expect(screen.getByText('Test Page')).toBeInTheDocument();
    expect(screen.getByText('Test subtitle')).toBeInTheDocument();
  });

  it('should render children content', () => {
    render(
      <PageLayout title='Test Page'>
        <div>Test Content</div>
      </PageLayout>
    );

    expect(screen.getByText('Test Content')).toBeInTheDocument();
  });
});

describe('PageLayout - Interactive Tests', () => {
  beforeEach(() => {
    mockUseSession.mockReturnValue({
      data: null,
      isPending: false,
      error: null,
      refetch: function (): void {
        throw new Error('Function not implemented.');
      },
      isRefetching: false,
    });
    mockUsePathname.mockReturnValue('/services');
  });

  it('should use breadcrumb override when provided', () => {
    const customBreadcrumbs = [
      { label: 'Custom', href: '/custom', isActive: false },
      { label: 'Path', href: '/custom/path', isActive: true },
    ];

    render(
      <PageLayout title='Test Page' breadcrumbOverride={customBreadcrumbs}>
        <div>Test Content</div>
      </PageLayout>
    );

    expect(screen.getByText('Custom')).toBeInTheDocument();
    expect(screen.getByText('Path')).toBeInTheDocument();
  });

  it('should apply custom className props', () => {
    const { container } = render(
      <PageLayout
        title='Test Page'
        className='custom-wrapper'
        headerClassName='custom-header'
        contentClassName='custom-content'
      >
        <div>Test Content</div>
      </PageLayout>
    );

    const wrapper = container.firstChild as HTMLElement;
    expect(wrapper).toHaveClass('custom-wrapper');
  });

  it('should toggle mobile auth dropdown', () => {
    render(
      <PageLayout title='Test Page'>
        <div>Test Content</div>
      </PageLayout>
    );

    const toggleButton = screen.getByLabelText('Toggle auth menu');
    fireEvent.click(toggleButton);

    const backdrop = document.querySelector('.fixed.inset-0');
    expect(backdrop).toBeInTheDocument();
  });

  it('should render without subtitle', () => {
    render(
      <PageLayout title='Test Page'>
        <div>Test Content</div>
      </PageLayout>
    );

    expect(screen.getByText('Test Page')).toBeInTheDocument();
    // Subtitle should not be rendered
    const subtitleElement = screen.queryByText('Test subtitle');
    expect(subtitleElement).not.toBeInTheDocument();
  });

  it('should render UserMenu when authenticated', () => {
    mockUseSession.mockReturnValue({
      data: {
        user: {
          id: '1',
          name: 'Test User',
          email: 'test@example.com',
          createdAt: new Date(),
          updatedAt: new Date(),
          emailVerified: false,
          twoFactorEnabled: undefined,
        },
        session: {
          id: '',
          createdAt: new Date(),
          updatedAt: new Date(),
          userId: '',
          expiresAt: new Date(),
          token: '',
        },
      },
      isPending: false,
      error: null,
      refetch: function (): void {
        throw new Error('Function not implemented.');
      },
      isRefetching: false,
    });

    render(
      <PageLayout title='Test Page'>
        <div>Test Content</div>
      </PageLayout>
    );

    // UserMenu should be rendered instead of auth buttons
    expect(screen.queryByText('Sign In')).not.toBeInTheDocument();
  });
});
