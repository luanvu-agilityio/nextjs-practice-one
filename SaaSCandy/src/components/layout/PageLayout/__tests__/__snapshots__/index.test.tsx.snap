// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PageLayout - Snapshot Tests should match snapshot 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="relative text-white "
      style="background-image: url("/images/background/heading-background.png"); background-size: cover;"
    >
      <header
        class="w-full relative z-10"
      >
        <div
          class="max-w-[1296px] mx-auto py-4 px-4 sm:px-6 lg:px-0 flex items-center gap-4 sm:gap-8 lg:gap-14"
        >
          <a
            class="flex items-center gap-2 no-underline flex-shrink-0"
            href="/"
          >
            <svg
              class="w-6 h-6 sm:w-8 sm:h-8"
              fill="none"
              height="32"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="15.9999"
                cy="15.9999"
                fill="#FFD245"
                r="15.9999"
              />
              <mask
                height="32"
                id="mask0_2053_384"
                maskUnits="userSpaceOnUse"
                style="mask-type: alpha;"
                width="32"
                x="0"
                y="0"
              >
                <circle
                  cx="15.9999"
                  cy="15.9999"
                  fill="#FFD245"
                  r="15.9999"
                />
              </mask>
              <g
                mask="url(#mask0_2053_384)"
              >
                <circle
                  cx="19.3045"
                  cy="9.53418"
                  fill="#FFE486"
                  r="15.9999"
                />
                <circle
                  cx="28.3808"
                  cy="9.53418"
                  fill="#EE7B11"
                  r="15.9999"
                />
              </g>
              <circle
                cx="24.7011"
                cy="24.7013"
                fill="#1642B9"
                r="6.29869"
                stroke="white"
                stroke-width="2"
              />
            </svg>
            <div
              class="hidden sm:block"
            >
              <h1
                aria-level="1"
                class="text-2xl font-semibold"
                role="heading"
              >
                <span
                  class="font-secondary text-primary text-xl sm:text-2xl lg:text-3xl font-light"
                >
                  SaaS
                  <span
                    class="font-medium"
                  >
                    Candy
                  </span>
                </span>
              </h1>
            </div>
          </a>
          <nav
            class="flex-1"
          >
            <ul
              class="hidden lg:flex items-center justify-start gap-6 xl:gap-10 text-primary w-full"
            >
              <li>
                <a
                  class="font-regular text-sm xl:text-lg hover:text-orange-background transition-colors "
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  class="font-regular text-sm xl:text-lg hover:text-orange-background transition-colors text-orange-background"
                  href="/services"
                >
                  Services
                </a>
              </li>
              <li>
                <a
                  class="font-regular text-sm xl:text-lg hover:text-orange-background transition-colors "
                  href="/about"
                >
                  About
                </a>
              </li>
              <li>
                <a
                  class="font-regular text-sm xl:text-lg hover:text-orange-background transition-colors "
                  href="/contact"
                >
                  Contact
                </a>
              </li>
              <li>
                <a
                  class="font-regular text-sm xl:text-lg hover:text-orange-background transition-colors "
                  href="/blog"
                >
                  Blog
                </a>
              </li>
            </ul>
            <div
              class="lg:hidden flex items-center"
            >
              <button
                aria-label="Toggle menu"
                class="cursor-pointer inline-flex items-center justify-center rounded-full transition-colors focus-visible:outline-none focus-visible:ring-0 focus-visible:ring-ring focus-visible:ring-offset-0 disabled:opacity-50 disabled:cursor-not-allowed button-animation bg-transparent border-none text-primary hover:opacity-90 w-6 h-6 p-0"
                data-testid="icon-button"
              >
                <svg
                  class="text-gray-700"
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M3 12h18M3 6h18M3 18h18"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </div>
          </nav>
          <div
            class="ml-auto flex items-center gap-2 sm:gap-4"
          >
            <svg
              class="w-8 h-8 sm:w-10 sm:h-10"
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="white"
                height="40"
                rx="20"
                width="40"
              />
              <path
                d="M20 30C25.523 30 30 25.523 30 20C30 19.537 29.306 19.46 29.067 19.857C28.5572 20.7013 27.862 21.4186 27.034 21.9545C26.206 22.4903 25.2669 22.8307 24.2878 22.9499C23.3088 23.0691 22.3154 22.9639 21.383 22.6423C20.4507 22.3207 19.6037 21.7911 18.9063 21.0937C18.2089 20.3963 17.6793 19.5493 17.3577 18.617C17.0361 17.6846 16.9309 16.6912 17.0501 15.7122C17.1693 14.7331 17.5097 13.794 18.0455 12.966C18.5814 12.138 19.2987 11.4428 20.143 10.933C20.54 10.693 20.463 10 20 10C14.477 10 10 14.477 10 20C10 25.523 14.477 30 20 30Z"
                fill="black"
              />
            </svg>
            <div
              class="hidden sm:flex items-center gap-2 sm:gap-4"
            >
              <a
                class="no-underline"
                href="/signin"
              >
                <button
                  class="inline-flex items-center justify-center font-semibold text-md leading-none tracking-normal rounded-xl transition-colors duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed cursor-pointer bg-white text-orange-background border border-orange-background hover:bg-orange-background hover:text-white focus-visible:ring-primary px-4 py-2.5"
                >
                  Sign In
                </button>
              </a>
              <a
                class="no-underline"
                href="/signup"
              >
                <button
                  class="inline-flex items-center justify-center font-semibold text-md leading-none tracking-normal rounded-xl transition-colors duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed cursor-pointer bg-orange-background text-white hover:opacity-90 focus-visible:ring-primary px-4 py-2.5"
                >
                  Sign Up
                </button>
              </a>
            </div>
            <div
              class="sm:hidden relative"
            >
              <button
                aria-label="Toggle auth menu"
                class="cursor-pointer inline-flex items-center justify-center rounded-full transition-colors focus-visible:outline-none focus-visible:ring-0 focus-visible:ring-ring focus-visible:ring-offset-0 disabled:opacity-50 disabled:cursor-not-allowed button-animation bg-transparent border-none hover:opacity-90 w-6 h-6 text-primary p-0"
                data-testid="icon-button"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-user w-6 h-6"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 21v-2a4 4 0 0 0-4-4H9a4 4 0 0 0-4 4v2"
                  />
                  <circle
                    cx="12"
                    cy="7"
                    r="4"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </header>
      <div
        class="relative flex flex-col lg:flex-row items-start lg:items-center justify-between max-w-[1296px] mx-auto px-4 sm:px-6 lg:px-0 pb-12 sm:pb-16 pt-12 sm:pt-20 lg:pt-29.25 gap-8"
      >
        <div
          class="space-y-4 sm:space-y-6 lg:space-y-8 w-full lg:w-auto"
        >
          <h2
            aria-level="2"
            class="text-primary font-bold text-3xl sm:text-4xl lg:text-5xl xl:text-6xl"
            role="heading"
          >
            Test Page
          </h2>
          <p
            class="text-lg text-gray-background leading-relaxed max-w-full lg:max-w-132 sm:text-lg"
            role="paragraph"
          >
            Test subtitle
          </p>
        </div>
        <nav
          aria-label="breadcrumb"
          class="flex items-center gap-1.5 text-sm text-slate-600 w-full lg:w-auto lg:flex-shrink-0"
        >
          <ol
            class="flex items-center gap-1.5 break-words text-sm text-muted-foreground sm:gap-2.5 flex-wrap"
          >
            <li
              class="inline-flex items-center gap-1.5 transition-colors hover:text-slate-900"
            >
              <a
                class="text-blue-foreground hover:text-white transition-colors text-sm sm:text-xs"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              aria-hidden="true"
              class="flex items-center text-white"
              role="presentation"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-right h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9 18 6-6-6-6"
                />
              </svg>
            </li>
            <li
              class="inline-flex items-center gap-1.5 transition-colors hover:text-slate-900"
            >
              <span
                aria-current="page"
                aria-disabled="true"
                class="text-white font-medium text-sm sm:text-xs"
                role="link"
              >
                Services
              </span>
            </li>
          </ol>
        </nav>
      </div>
    </div>
    <div
      class="bg-inactive-background "
    >
      <div>
        Test Content
      </div>
    </div>
  </div>
</div>
`;
